src/lib/es2020.intl.d.ts

```diff
@@ -313,7 +308,7 @@ declare namespace Intl {
      * [MDN](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Intl/Locale).
      */
     const Locale: {
-        new (tag: BCP47LanguageTag | Locale, options?: LocaleOptions): Locale;
+        new(tag: UnicodeBCP47LocaleIdentifier | Locale, options?: LocaleOptions): Locale;
     };
 
     type DisplayNamesFallback =

```

Comment:
Arguably [`DisplayNames#of`](https://tc39.es/ecma402/#sec-Intl.DisplayNames.prototype.of)'s `code` argument should be widened to accept `string | Intl.Locale` if (and only if) the `type` of the `DisplayNames` instance is `language`, but that would require making `DisplayNames` generic.

For `DisplayNames#of`, the spec doesn't check the `[[Locale]]` internal slot, rather it calls `toString` (identical result for `Intl.Locale` objects) followed by checking `IsStructurallyValidLanguageTag`, which verifies the string matches a `unicode_locale_id`, and which is always true for `Intl.Locale` objects' string representation.

Would this be considered a valid change to make given that the spec guarantees it will never throw at runtime? And would making `DisplayNames` generic be acceptable for that purpose?